src/envoy/utils/utils_test.cc

```diff
@@ -128,6 +128,11 @@ TEST_P(UtilsTest, GetPrincipalNoSpiffePrefix) {
   testGetPrincipal(sans, "spiffe:foo/bar", true);
 }
 
+TEST_P(UtilsTest, GetPrincipalFromSpiffePrefixSAN) {
+  std::vector<std::string> sans{"bad", "spiffe://foo/bar"};
+  testGetPrincipal(sans, "foo/bar", true);
+}
+
```

Comment:
For completeness,  can you also add similar test cases for GetTrustDomain below.

Reply:
Added tests for GetTrustDomain as well, thank you.